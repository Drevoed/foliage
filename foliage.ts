export const css: any = () => null;
export const keyframes: any = () => null;
export const createGlobalStyle: any = () => null;

export function assertSelector(a) {
  return `.${a.css}`;
}
