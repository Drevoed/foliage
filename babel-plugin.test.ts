import pluginTester from 'babel-plugin-tester';
import plugin from './babel-plugin';

const fullExample = `
import { css, keyframes, createGlobalStyle } from 'foliage';
const part = css\`
  color: black;
  & > * { padding-left: 1rem; }
\`

const anim = keyframes\`
  0% { transform: rotate(0deg) }
  100% { transform: rotate(180deg) }
\`

const glob = createGlobalStyle\`
  body {
    position: fixed;
    top: 0;
    left: 0;
  }
\`
`;

const namespaceSupport = `
import * as example from 'foliage';
const part = example.css\`
  color: black;
  & > * { padding-left: 1rem; }
\`

const anim = example.keyframes\`
  0% { transform: rotate(0deg) }
  100% { transform: rotate(180deg) }
\`

const glob = example.createGlobalStyle\`
  body {
    position: fixed;
    top: 0;
    left: 0;
  }
\`

const theme = example.vars\`
  --demo: 123px;
\`
`;

const doNotCompilesThirdParty = `
import { css, keyframes, createGlobalStyle } from 'styled-components';
const part = css\`color:black;\`
const anim = keyframes\`from{color: black;} to {color: white;}\`
const glob = createGlobalStyle\`body{display: grid}\`
`;

const cssInterpolationInCss = `
import { css } from 'foliage';
const first = css\`
  color: black;
  & > * { padding-left: 1rem; }
\`
const second = css\`
  color: red;
  & \${first} {
    color: white;
  }
\`
`;

const keyframesInterpolationInCss = `
import { css } from 'foliage';
const anim = keyframes\`
  0% { transform: rotate(0deg) }
  100% { transform: rotate(180deg) }
\`
const element = css\`
  color: red;
  animation: \${anim} 3s infinite;
\`
`;

pluginTester({
  pluginName: 'foliage',
  plugin,
  pluginOptions: { debug: true },
  root: __dirname,
  filename: __filename,
  babelOptions: { filename: __filename },
  snapshot: true,
  tests: {
    cssImportedFromPackage: `import {css} from 'foliage'; const example = css\`color: black\``,
    cssRenamedFromPackage: `import {css as demo} from 'foliage'; const example = demo\`justify-content: center;\``,
    withoutDebugNamesNotAdded: {
      code: `import {css} from 'foliage'; const example = css\`span{display:grid}\``,
      pluginOptions: { debug: false },
    },
    cssImportedFromReact: `import {css} from 'foliage-react'; const demo = css\`.global & { [data-demo="a"] {color: black} }\``,
    keyframesAndGlobalStylesShouldCompile: fullExample,
    varsNotCompiles: `import {vars, css} from 'foliage'; const a = css\`outline: none;\`; const b = vars\`--random: 1px;\``,
    namespaceSupport,
    doNotCompilesThirdParty,
    cssInterpolationInCss,
    keyframesInterpolationInCss,
    prefixAdd: {
      code: fullExample,
      pluginOptions: { prefix: 'prefix', debug: true },
    },
    prefixExistsWithoutDebug: {
      code: fullExample,
      pluginOptions: { prefix: 'prefix', debug: false },
    },
  },
});
