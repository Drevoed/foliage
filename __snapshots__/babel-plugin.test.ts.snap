// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`foliage cssImportedFromPackage: cssImportedFromPackage 1`] = `

import {css} from 'foliage'; const example = css\`color: black\`

      ↓ ↓ ↓ ↓ ↓ ↓

import { css } from 'foliage';
const example = {
  content: '.xb1pwo-example{color:#000}',
  css: 'xb1pwo-example',
};


`;

exports[`foliage cssImportedFromReact: cssImportedFromReact 1`] = `

import {css} from 'foliage-react'; const demo = css\`.global & { [data-demo="a"] {color: black} }\`

      ↓ ↓ ↓ ↓ ↓ ↓

import { css } from 'foliage-react';
const demo = {
  content: '.global .csqqgc-demo [data-demo=a]{color:#000}',
  css: 'csqqgc-demo',
};


`;

exports[`foliage cssInterpolationInCss: cssInterpolationInCss 1`] = `

import { css } from 'foliage';
const first = css\`
  color: black;
  & > * { padding-left: 1rem; }
\`
const second = css\`
  color: red;
  & \${first} {
    color: white;
  }
\`

      ↓ ↓ ↓ ↓ ↓ ↓

import { css } from 'foliage';
const first = {
  content: '.yuc9wf-first{color:#000}.yuc9wf-first>*{padding-left:1rem}',
  css: 'yuc9wf-first',
};
const second = {
  content: 'COMPILED CSS HERE',
  css: 'a8hqie-second',
};


`;

exports[`foliage cssRenamedFromPackage: cssRenamedFromPackage 1`] = `

import {css as demo} from 'foliage'; const example = demo\`justify-content: center;\`

      ↓ ↓ ↓ ↓ ↓ ↓

import { css as demo } from 'foliage';
const example = {
  content: '.xb1qln-example{justify-content:center}',
  css: 'xb1qln-example',
};


`;

exports[`foliage doNotCompilesThirdParty: doNotCompilesThirdParty 1`] = `

import { css, keyframes, createGlobalStyle } from 'styled-components';
const part = css\`color:black;\`
const anim = keyframes\`from{color: black;} to {color: white;}\`
const glob = createGlobalStyle\`body{display: grid}\`

      ↓ ↓ ↓ ↓ ↓ ↓

import { css, keyframes, createGlobalStyle } from 'styled-components';
const part = css\`
  color: black;
\`;
const anim = keyframes\`from{color: black;} to {color: white;}\`;
const glob = createGlobalStyle\`body{display: grid}\`;


`;

exports[`foliage keyframesAndGlobalStylesShouldCompile: keyframesAndGlobalStylesShouldCompile 1`] = `

import { css, keyframes, createGlobalStyle } from 'foliage';
const part = css\`
  color: black;
  & > * { padding-left: 1rem; }
\`

const anim = keyframes\`
  0% { transform: rotate(0deg) }
  100% { transform: rotate(180deg) }
\`

const glob = createGlobalStyle\`
  body {
    position: fixed;
    top: 0;
    left: 0;
  }
\`

      ↓ ↓ ↓ ↓ ↓ ↓

import { css, keyframes, createGlobalStyle } from 'foliage';
const part = {
  content: '.pr0ig0-part{color:#000}.pr0ig0-part>*{padding-left:1rem}',
  css: 'pr0ig0-part',
};
const anim = {
  content:
    '@-webkit-keyframes wpxzoa-anim{0%{transform:rotate(0deg)}to{transform:rotate(180deg)}}@keyframes wpxzoa-anim{0%{transform:rotate(0deg)}to{transform:rotate(180deg)}}',
  keyframes: 'wpxzoa-anim',
};
const glob = {
  content: 'body{position:fixed;top:0;left:0}',
  createGlobalStyle: 'l0hjnn-glob',
};


`;

exports[`foliage keyframesInterpolationInCss: keyframesInterpolationInCss 1`] = `

import { css } from 'foliage';
const anim = keyframes\`
  0% { transform: rotate(0deg) }
  100% { transform: rotate(180deg) }
\`
const element = css\`
  color: red;
  animation: \${anim} 3s infinite;
\`

      ↓ ↓ ↓ ↓ ↓ ↓

import { css } from 'foliage';
const anim = keyframes\`
  0% { transform: rotate(0deg) }
  100% { transform: rotate(180deg) }
\`;
const element = {
  content: 'COMPILED CSS HERE',
  css: '70rkv4-element',
};


`;

exports[`foliage namespaceSupport: namespaceSupport 1`] = `

import * as example from 'foliage';
const part = example.css\`
  color: black;
  & > * { padding-left: 1rem; }
\`

const anim = example.keyframes\`
  0% { transform: rotate(0deg) }
  100% { transform: rotate(180deg) }
\`

const glob = example.createGlobalStyle\`
  body {
    position: fixed;
    top: 0;
    left: 0;
  }
\`

      ↓ ↓ ↓ ↓ ↓ ↓

import * as example from 'foliage';
const part = example.css\`
  color: black;
  & > * { padding-left: 1rem; }
\`;
const anim = example.keyframes\`
  0% { transform: rotate(0deg) }
  100% { transform: rotate(180deg) }
\`;
const glob = example.createGlobalStyle\`
  body {
    position: fixed;
    top: 0;
    left: 0;
  }
\`;


`;

exports[`foliage prefixAdd: prefixAdd 1`] = `

import { css, keyframes, createGlobalStyle } from 'foliage';
const part = css\`
  color: black;
  & > * { padding-left: 1rem; }
\`

const anim = keyframes\`
  0% { transform: rotate(0deg) }
  100% { transform: rotate(180deg) }
\`

const glob = createGlobalStyle\`
  body {
    position: fixed;
    top: 0;
    left: 0;
  }
\`

      ↓ ↓ ↓ ↓ ↓ ↓

import { css, keyframes, createGlobalStyle } from 'foliage';
const part = {
  content:
    '.prefix-pr0ig0-part{color:#000}.prefix-pr0ig0-part>*{padding-left:1rem}',
  css: 'prefix-pr0ig0-part',
};
const anim = {
  content:
    '@-webkit-keyframes prefix-wpxzoa-anim{0%{transform:rotate(0deg)}to{transform:rotate(180deg)}}@keyframes prefix-wpxzoa-anim{0%{transform:rotate(0deg)}to{transform:rotate(180deg)}}',
  keyframes: 'prefix-wpxzoa-anim',
};
const glob = {
  content: 'body{position:fixed;top:0;left:0}',
  createGlobalStyle: 'prefix-l0hjnn-glob',
};


`;

exports[`foliage prefixExistsWithoutDebug: prefixExistsWithoutDebug 1`] = `

import { css, keyframes, createGlobalStyle } from 'foliage';
const part = css\`
  color: black;
  & > * { padding-left: 1rem; }
\`

const anim = keyframes\`
  0% { transform: rotate(0deg) }
  100% { transform: rotate(180deg) }
\`

const glob = createGlobalStyle\`
  body {
    position: fixed;
    top: 0;
    left: 0;
  }
\`

      ↓ ↓ ↓ ↓ ↓ ↓

import { css, keyframes, createGlobalStyle } from 'foliage';
const part = {
  content: '.prefix-pr0ig0{color:#000}.prefix-pr0ig0>*{padding-left:1rem}',
  css: 'prefix-pr0ig0',
};
const anim = {
  content:
    '@-webkit-keyframes prefix-wpxzoa{0%{transform:rotate(0deg)}to{transform:rotate(180deg)}}@keyframes prefix-wpxzoa{0%{transform:rotate(0deg)}to{transform:rotate(180deg)}}',
  keyframes: 'prefix-wpxzoa',
};
const glob = {
  content: 'body{position:fixed;top:0;left:0}',
  createGlobalStyle: 'prefix-l0hjnn',
};


`;

exports[`foliage varsNotCompiles: varsNotCompiles 1`] = `

import {vars, css} from 'foliage'; const a = css\`outline: none;\`; const b = vars\`--random: 1px;\`

      ↓ ↓ ↓ ↓ ↓ ↓

import { vars, css } from 'foliage';
const a = {
  content: '.fo0rip-a{outline:0}',
  css: 'fo0rip-a',
};
const b = vars\`--random: 1px;\`;


`;

exports[`foliage withoutDebugNamesNotAdded: withoutDebugNamesNotAdded 1`] = `

import {css} from 'foliage'; const example = css\`span{display:grid}\`

      ↓ ↓ ↓ ↓ ↓ ↓

import { css } from 'foliage';
const example = {
  content: '.xb1pwo span{display:grid}',
  css: 'xb1pwo',
};


`;
